components/Common/Pagination/index.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import { ArrowRightIcon, ArrowLeftIcon } from '@heroicons/react/20/solid';
+import type { FC } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import Button from '@/components/Common/Button';
+
+import styles from './index.module.css';
+import { useGetPageElements } from './useGetPageElements';
+
+type Page = {
+  url: string;
+};
+
+export type PaginationProps = {
+  /**
+   * One-based number of the current page
+   */
+  currentPage: number;
+  pages: Page[];
+  /**
+   * The number of page buttons on each side of the current page button
+   * @default 1
+   */
+  currentPageSiblingsCount?: number;
+};
+
+const Pagination: FC<PaginationProps> = ({
+  currentPage,
+  pages,
+  currentPageSiblingsCount = 1,
+}) => {
+  const intl = useIntl();
+
+  const parsedPages = useGetPageElements(
+    currentPage,
+    pages,
+    currentPageSiblingsCount
+  );
+
+  return (
+    <nav
+      aria-label={intl.formatMessage({
+        id: 'components.common.pagination.defaultLabel',
+      })}
+      className={styles.pagination}
+    >
+      <Button
+        type="button"
+        aria-label={intl.formatMessage({
+          id: 'components.common.pagination.prevAriaLabel',
+        })}
+        disabled={currentPage === 1}
+        variant="secondary"
+        className={styles.previousButton}
+      >
+        <ArrowLeftIcon className={styles.arrowIcon} />
+        <span>
```

Comment:
Is a span needed here? Afaik you can directly on `FormattedMessage` specify it to use a "span" element. So no need to surround with yet another span.

Reply:
I did not know that `FormattedMessage` added a span already. I will definitely remove the extra `span` element, then. Thank you

Reply:
I have checked, and `FormattedMessage` seems to just add the text by default, so no extra `span` was being added here.

<img width="890" alt="image" src="https://github.com/nodejs/nodejs.org/assets/93147810/ef121d8f-67b0-4123-8f0d-d2d4d15748a0">

It only enclosed the translation string within a tag when adding the `tagName` prop, so I can remove `span` and just pass it through the `tagName` property if that sounds better.